src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs

```diff
@@ -224,37 +177,19 @@ public async Task<bool> IsServerLicenseValid()
             }
         }
 
-        private static SonarRule FilterRule(JObject r, JToken actives)
+        private static SonarRule CreateRule(JObject r, JToken actives)
         {
-            var activeRulesForRuleKey = actives.Value<JArray>(r["key"].ToString());
-
-            if (activeRulesForRuleKey == null ||
-                activeRulesForRuleKey.Count != 1)
-            {
-                // Because of the parameters we use we expect to have only actives rules. So rules and actives
-                // should both contain the same number of elements (i.e. the same rules).
-                throw new JsonException($"Malformed json response, \"actives\" field should contain rule '{r["key"].ToString()}'");
-            }
-
-            var activeRule = new SonarRule(r["repo"].ToString(), ParseRuleKey(r["key"].ToString()), true);
-            if (r["internalKey"] != null)
-            {
-                activeRule.InternalKey = r["internalKey"].ToString();
-            }
-            if (r["templateKey"] != null)
+            var active = actives?.Value<JArray>(r["key"].ToString())?.FirstOrDefault();
+            var rule = new SonarRule(r["repo"].ToString(), ParseRuleKey(r["key"].ToString()), r["internalKey"]?.ToString(), r["templateKey"]?.ToString(), active != null);
```

Comment:
Would be great to extract the strings as constants.

Reply:
It's used just once.

Reply:
There are some which are used multiple times e.g. "key"